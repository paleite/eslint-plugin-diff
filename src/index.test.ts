process.env.CI = "true";
import * as child_process from "child_process";

jest.mock("child_process");
const mockedChildProcess = jest.mocked(child_process, { shallow: true });
mockedChildProcess.execFileSync.mockReturnValue("line1\nline2\nline3");

import { configs, processors } from "./index";

describe("plugin", () => {
  it("should match expected export", () => {
    expect(configs).toMatchSnapshot();
    expect(processors).toMatchSnapshot();
  });
});
