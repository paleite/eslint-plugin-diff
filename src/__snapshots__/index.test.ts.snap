// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin should match expected export 1`] = `
{
  "ci": {
    "overrides": [
      {
        "files": [
          "*",
        ],
        "processor": "diff/ci",
      },
    ],
    "plugins": [
      "diff",
    ],
  },
  "committed": {
    "overrides": [
      {
        "files": [
          "*",
        ],
        "processor": "diff/committed",
      },
    ],
    "plugins": [
      "diff",
    ],
  },
  "diff": {
    "overrides": [
      {
        "files": [
          "*",
        ],
        "processor": "diff/diff",
      },
    ],
    "plugins": [
      "diff",
    ],
  },
  "staged": {
    "overrides": [
      {
        "files": [
          "*",
        ],
        "processor": "diff/staged",
      },
    ],
    "plugins": [
      "diff",
    ],
  },
}
`;

exports[`plugin should match expected export 2`] = `
{
  "ci": {
    "postprocess": [Function],
    "preprocess": [Function],
    "supportsAutofix": true,
  },
  "committed": {
    "postprocess": [Function],
    "preprocess": [Function],
    "supportsAutofix": false,
  },
  "diff": {
    "postprocess": [Function],
    "preprocess": [Function],
    "supportsAutofix": true,
  },
  "staged": {
    "postprocess": [Function],
    "preprocess": [Function],
    "supportsAutofix": true,
  },
}
`;
